### PR TITLE
Fix location field not exported correctly to office 2007 xml

### PR DESCRIPTION
**Pull request by Siedlerchr**
Fix location field not exported correctly to office 2007 xml

- [X] Change in CHANGELOG.md described
- [X] Manually tested changed features in running JabRef
- [X] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)

### CODE REVIEW

File: src/main/java/net/sf/jabref/logic/msbib/MSBibEntry.java [hunk 1]
```diff
@@ -65,10 +65,11 @@
...
     // reduced subset, supports only "CITY , STATE, COUNTRY"
-    // \b(\w+)\s?[,]?\s?(\w+)\s?[,]?\s?(\w+)\b
-    // WORD SPACE , SPACE WORD SPACE , SPACE WORD
-    // tested using http://www.javaregex.com/test.html
-    private static final Pattern ADDRESS_PATTERN = Pattern.compile("\\b(\\w+)\\s?[,]?\\s?(\\w+)\\s?[,]?\\s?(\\w+)\\b");
+    // \b(\w+)\s?[,]?\s?(\w+)\s?[,]?\s?(\w*)\b
+    // WORD SPACE , SPACE WORD SPACE (zero or more) , SPACE WORD (zero or word)
+    // tested using http://www.regexpal.com/
+    //Matches both single locations like Berlin or Stroudsburg, PA, USA
+    private static final Pattern ADDRESS_PATTERN = Pattern.compile("\\b(\\w+)\\s?[,]?\\s?(\\w*)\\s?[,]?\\s?(\\w*)\\b");
```
Comment (stefan-kolb) [inline src/main/java/net/sf/jabref/logic/msbib/MSBibEntry.java]: Maybe we can think of documenting our regexes more. See DOI.java for an example.

**Comment by stefan-kolb** [comment]
Can you please add tests for this?

**Comment by Siedlerchr** [comment]
Yeah, already planned this. Just wanted to create a first working version.

**Comment by Siedlerchr** [comment]
Added export/import test

File: src/main/java/net/sf/jabref/logic/msbib/MSBibEntry.java [hunk 2]
```diff
@@ -64,25 +64,32 @@
...
-    // Allows 20.3-2007|||20/3-  2007 etc.
-    // (\d{1,2})\s?[.,-/]\s?(\d{1,2})\s?[.,-/]\s?(\d{2,4})
-    // 1-2 DIGITS SPACE SEPERATOR SPACE 1-2 DIGITS SPACE SEPERATOR SPACE 2-4 DIGITS
-    // tested using http://www.javaregex.com/test.html
+    /**
+     * reduced subset, supports only "CITY , STATE, COUNTRY" <br>
+     *  <b>\b(\w+)\s?[,]?\s?(\w+)\s?[,]?\s?(\w*)\b</b> <br>
+     *  WORD SPACE , SPACE WORD SPACE (Can be zero or more) , SPACE WORD (Can be zero or more) <br>
+     *  Matches both single locations (only city) like Berlin and full locations like Stroudsburg, PA, USA <br>
+     *  tested using http://www.regexpal.com/
```
Comment (stefan-kolb) [inline src/main/java/net/sf/jabref/logic/msbib/MSBibEntry.java]: Link is not relevant IMHO

File: src/main/java/net/sf/jabref/logic/msbib/MSBibEntry.java [hunk 3]
```diff
@@ -64,25 +64,32 @@
...
+     * <b>(\d{1,2})\s?[.,-/]\s?(\d{1,2})\s?[.,-/]\s?(\d{2,4})</b>
+     * 1-2 DIGITS SPACE SEPERATOR SPACE 1-2 DIGITS SPACE SEPERATOR SPACE 2-4 DIGITS
+     */
     private static final Pattern DATE_PATTERN = Pattern
             .compile("(\\d{1,2})\\s*[.,-/]\\s*(\\d{1,2})\\s*[.,-/]\\s*(\\d{2,4})");
 
 
     public MSBibEntry() {
-
+        //empty
```
Comment (stefan-kolb) [inline src/main/java/net/sf/jabref/logic/msbib/MSBibEntry.java]: Can be removed.

**Comment by stefan-kolb** [comment]
LGTM :+1:  Please merge.

File: src/main/java/net/sf/jabref/logic/msbib/MSBibEntry.java (hunk 3 above)
Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/logic/msbib/MSBibEntry.java]: Codacy and PMD has this as pattern ( UncommentedEmptyConstructor) to distinguish between internationally and unintentionally empty constructors
http://pmd.sourceforge.net/pmd-5.0.1/rules/java/design.html

**Pull request merged**